File: Search/BinarySearch.js
function binarySearch(arr, target) {
    let n = arr.length;
    let start = 0;
    let end = n - 1;
    
    while (start <= end) {
        let mid = Math.floor((start + end) / 2);

        if (arr[mid] === target) {
            return mid;
        }

        if (arr[mid] < target) {
            start = mid + 1;
        }

        else {
            end = mid - 1;
        }
    }
    return -1;
}

let array = [2, 3, 4, 10, 40];
let target = 10;
console.log(binarySearch(array, target));

File: Search/BinarySearch.c
#include <stdio.h>

int binarySearch(int arr[], int n, int target) {
  int left = 0;
  int right = n - 1;

  while (left <= right) {
    int mid = ((left + right) / 2);

    if (arr[mid] == target) {
        return mid;
    }

    if (arr[mid] < target) {
        left = mid + 1;
    } else {
        right = mid - 1;
    }
  }
  return -1;
}

int main() {
    int arr[] = {11, 12, 22, 25, 34, 64, 90};
    int target = 22;
    int n = sizeof(arr) / sizeof(arr[0]);
    int index = binarySearch(arr, n, target);

    if (index == -1) {
        printf("Element not found\n");
    } else {
        printf("Element found at index %d\n", index);
    }

    return 0;
}